api/env.mli

```diff
@@ -36,85 +29,93 @@ val set_debug_mode : string -> unit
     May raise DebugFlagNotRecognized.
 *)
 
+(**{2 Utilities} *)
+
 val check_arity : bool ref
 (** Flag to check for variables arity. Default is true. *)
 
 val check_ll : bool ref
-(** Flag to check for rules left linearity. Default is false. *)
+(** Flag to check for rules left linearity. Default is false *)
 
 (** {2 The Global Environment} *)
-module type S =
-sig
-  module Printer : Pp.Printer
-  val raise_env : loc -> env_error -> 'a
 
-  val init        : string -> mident
-  (** [init name] initializes a new global environement giving it the name of
-      the corresponding source file. The function returns the module identifier
-      corresponding to this file, built from its basename. Every toplevel
-      declaration will be qualified by this name. *)
+val init        : Parser.t -> t
+(** [init input] initializes a new global environement from the [input] *)
+
+val get_input    : t -> Parser.t
+(** [get_input env] returns the input used to create [env] *)
+
+val get_filename : t -> string
+(** [get_input env] returns the filename associated to the input of [env]. We return a fake filename if the input was not create from a filename. *)
+
+val get_signature : t -> Signature.t
+(** [get_signature env] returns the signature used by this module. *)
+
+val get_name    : t -> mident
+(** [get_name env] returns the name of the module. *)
 
-  val get_signature : unit -> Signature.t
-  (** [get_signature ()] returns the signature used by this module. *)
+val set_reduction_engine : t -> (module Reduction.S) -> t
+(** [set_reduction_egine env] changes the reduction engine of [env]. The new environment shares the same signature than [env]. *)
 
-  val get_name    : unit -> mident
-  (** [get_name ()] returns the name of the module. *)
+val get_reduction_engine : t -> (module Reduction.S)
+(** [get_reduction_engine env] returns the reduction engine of [env] *)
 
-  module HName : Hashtbl.S with type key = name
+val get_printer : t -> (module Pp.Printer)
+(** [get_print env] returns a pretty printer associated to [env] *)
 
-  val get_symbols : unit -> Signature.rw_infos HName.t
-  (** [get_symbols ()] returns the content of the signature [sg]. *)
+module HName : Hashtbl.S with type key = name
 
-  val get_type    : loc -> name -> term
-  (** [get_type l md id] returns the type of the constant [md.id]. *)
+val get_symbols : t -> Signature.rw_infos HName.t
+(** [get_symbols env] returns the content of the signature [sg]. *)
```

Comment:
Could you describe what is bound to a key in the returned hashtable (I guess a symbol).

Reply:
It is obvous from the type right? I've added something though.